src/pages/sponsors_inventory/popup/sponsor-inventory-popup.js

```diff
@@ -112,12 +112,29 @@ const SponsorItemDialog = ({
 
     if (!isConfirmed) return;
 
+    const removeOrResetField = () => {
+      if (formik.values.meta_fields.length === 1) {
+        formik.setFieldValue("meta_fields", [
+          {
+            name: "",
+            type: "Text",
+            is_required: false,
+            minimum_quantity: 0,
+            maximum_quantity: 0,
+            values: []
+          }
+        ]);
+      } else {
+        removeFormik(index);
+      }
+    };
+
     if (fieldType.id) {
       onMetaFieldTypeDeleted(initialEntity.id, fieldType.id)
-        .then(() => removeFormik(index))
+        .then(() => removeOrResetField(index))
```

Comment:
@tomrndom the function defnition does not no accepts any index param
const removeOrResetField = () => {